pagerduty/resource_pagerduty_extension_servicenow_test.go

```diff
@@ -0,0 +1,193 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckPagerDutyExtensionServicenowExists("pagerduty_extension_servicenow.foo"),
+					resource.TestCheckResourceAttr(
+						"pagerduty_extension_servicenow.foo", "name", extension_name_updated),
+					resource.TestCheckResourceAttr(
+						"pagerduty_extension_servicenow.foo", "extension_schema", "PJFWPEP"),
+					resource.TestCheckResourceAttr(
+						"pagerduty_extension_servicenow.foo", "endpoint_url", url_updated),
+					resource.TestCheckResourceAttr(
+						"pagerduty_extension_servicenow.foo", "config", "{\"notify_types\":{\"acknowledge\":true,\"assignments\":true,\"resolve\":true},\"restrict\":\"pd-users\"}"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckPagerDutyExtensionServicenowDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*pagerduty.Client)
+	for _, r := range s.RootModule().Resources {
+		if r.Type != "pagerduty_extension_servicenow" {
+			continue
+		}
+
+		if _, _, err := client.Extensions.Get(r.Primary.ID); err == nil {
+			return fmt.Errorf("Extension still exists")
+		}
+
+	}
+	return nil
+}
+
+func testAccCheckPagerDutyExtensionServicenowExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No extension ID is set")
+		}
+
+		client := testAccProvider.Meta().(*pagerduty.Client)
+
+		found, _, err := client.Extensions.Get(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if found.ID != rs.Primary.ID {
+			return fmt.Errorf("Extension not found: %v - %v", rs.Primary.ID, found)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckPagerDutyExtensionServicenowConfig(name string, extension_name string, url string, notify_types string, restrict string) string {
+	return fmt.Sprintf(`
+resource "pagerduty_user" "foo" {
+  name        = "%[1]v"
+  email       = "%[1]v@foo.com"
+  color       = "green"
+  role        = "user"
+  job_title   = "foo"
+  description = "foo"
+}
+
+resource "pagerduty_escalation_policy" "foo" {
+  name        = "%[1]v"
+  description = "bar"
+  num_loops   = 2
+
+  rule {
+    escalation_delay_in_minutes = 10
+
+    target {
+      type = "user_reference"
+      id   = pagerduty_user.foo.id
+    }
+  }
+}
+
+resource "pagerduty_service" "foo" {
+  name                    = "%[1]v"
+  description             = "foo"
+  auto_resolve_timeout    = 1800
+  acknowledgement_timeout = 1800
+  escalation_policy       = pagerduty_escalation_policy.foo.id
+
+  incident_urgency_rule {
+    type    = "constant"
+    urgency = "high"
+  }
+}
+
+data "pagerduty_extension_schema" "foo" {
+	name = "Generic V2 Webhook"
+}
+
+resource "pagerduty_extension_servicenow" "foo"{
+  name = "%s"
+  endpoint_url = "%s"
+  extension_schema = data.pagerduty_extension_servicenow_schema.foo.id
```

Comment:
I think this was supposed to be referencing `data.pagerduty_extension_schema.foo.id`